### PR TITLE
[clang-tidy] readability-avoid-const-params-in-decls

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: include/openthread/coap.h [hunk 1]
```diff
@@ -747,7 +747,7 @@ const otCoapOption *otCoapOptionIteratorGetNextOption(otCoapOptionIterator *aIte
  *
  * @see otCoapMessageAppendUintOption
  */
-otError otCoapOptionIteratorGetOptionUintValue(otCoapOptionIterator *aIterator, uint64_t *const aValue);
+otError otCoapOptionIteratorGetOptionUintValue(otCoapOptionIterator *aIterator, uint64_t *aValue);
```
Comment (wgtdkp) [inline include/openthread/coap.h:750]: should we also change the implementation signature ([here](https://github.com/openthread/openthread/blob/12b967cce2e06840dd9bccc67971b617c8f43a3c/src/core/api/coap_api.cpp#L207)) to keep them consistent

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by wgtdkp — APPROVED** [review]
LGTM 👍

**Pull request merged**